src/bodies.cpp

```diff
@@ -535,7 +535,7 @@ void bodies::Box::updateInternalData()
   normalW_ = basis.col(1);
   normalH_ = basis.col(2);
 
-  const Eigen::Vector3d tmp(normalL_ * length2_ + normalW_ * width2_ + normalH_ * height2_);
+  const Eigen::Vector3d tmp(length2_, width2_, height2_);
```

Comment:
Obviously this changes the semantics of `corner1_` and `corner2_`. Could you please document this in the header? Does this change have implications for other pieces of the code?

Could we remove the helper variables normalL, normalW, normalH?
These are only used in Box::containsPoint(), which could be simplified to:
```
const Eigen::Vector3d aligned = (pose_.rotation().transpose() * (p - center_)).abs();
return aligned[0] <= length2_ && aligned[1] <= width2_ && aligned[2] <= height2_;
```

Reply:
Ok, I'll have a look at this.

Reply:
I think you definitely don't want to call `rotation()` in `containsPoint` as this function calls SVD if I'm not wrong. The question is if calling `linear()` wouldn't do the same job here (which I think would).

Changing the semantics of `corner1_` and `corner2_` wouldn't change anything in this library except for the ray-box test. The variables are protected, so I think chances that somebody else would be using them is low. But maybe it'd be better to rename them so that people who were using these variables will get a build failure. Or I can leave the old variables there and add new ones, but as the old ones were semantically wrong, I'm more for the option of getting rid of them.

Reply:
> I think you definitely don't want to call rotation() in containsPoint as this function calls SVD if I'm not wrong.

It was calling SVD, when we used Affine3d, but since we use Isometry3d, Eigen knows thatit can just return the 3x3 rotation sub-matrix. No overhead at all.

> Changing the semantics of corner1_ and corner2_ wouldn't change anything in this library.

I just wanted to be sure. No need to change the variable names. Just document their semantics in the header (there was nothing before).